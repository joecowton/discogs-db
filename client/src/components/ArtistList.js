import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import * as actions from '../actions';

class ReleaseList extends Component {
  componentWillMount() {
    this.props.fetchArtist("Kowton");
    // this.props.fetchData();
  }

  handleClick() {
  }

  renderData(data) {
    return (
        <ul key={data.id} className="collection">
          <li className="collection-item avatar">
            <img src={data.thumb} alt="" className="circle"></img>
            <span className="title">{data.artist}</span>
            <p>{data.title} <br />
               {data.label}
            </p>
            <Link to={`/release/${data.id}`} className="secondary-content"><i className="material-icons">View</i></Link>
          </li>
        </ul>
    )
  }

  render () {
    if (!this.props.data.length){
      return <div>Loading..</div>
    }

      return(
        <div className="release-list">
          {this.props.data.map(this.renderData)}
        </div>
      )

  }
}

function mapStateToProps(state) {
  return { data: state.data }
}

export default connect(mapStateToProps, actions)(ReleaseList)
