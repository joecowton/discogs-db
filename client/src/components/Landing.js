import React from 'react';

const Landing = () => {
  return(
    <div class="jumbotron center">
      <h1>Videos!</h1>

      <p><a class="btn btn-primary btn-lg" href="/auth/google" role="button">Log in</a></p>
    </div>
  )
}

export default Landing;
