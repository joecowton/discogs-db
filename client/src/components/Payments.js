import React, { Component } from 'react';
import StripeCheckout from 'react-stripe-checkout';
import { connect } from 'react-redux';
import * as actions from '../actions';

class Payments extends Component {
  render (){
     const { lowest_price } = this.props.amount
     console.log(lowest_price);
     return (
      <StripeCheckout
        name="DISCOGS"
        description={lowest_price}
        amount={lowest_price}
        token={token => this.props.handleToken(token)}
        stripeKey={process.env.REACT_APP_STRIPE_KEY}
      >
          <button className="btn black">
            Â£{lowest_price}
          </button>
      </StripeCheckout>
    );
  }
}

export default connect(null, actions)(Payments);
