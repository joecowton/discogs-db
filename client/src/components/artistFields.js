export default [
	{ searchName: 'Kowton', displayName: 'Kowton' },
	{ searchName: 'Pev*', displayName: 'Peverelist' },
	{ searchName: 'Forest Drive West', displayName: 'FDW' },
	{ searchName: 'Asusu', displayName: 'Asusu' },
	{ searchName: 'Hodge', displayName: 'Hodge' },
	{ searchName: 'Various', displayName: 'Various' }
];
