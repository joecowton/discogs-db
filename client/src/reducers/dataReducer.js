import {
    FETCH_DATA,
    FETCH_RELEASE,
    FETCH_ARTIST,
    SUBMIT_RELEASE,
} from '../actions/types';

export default function(state: Array<Object> = [], action) {
    switch (action.type) {
        case FETCH_DATA:
            return [...state, ...action.payload];
        case FETCH_ARTIST:
            return action.payload;
        case SUBMIT_RELEASE:
            return action.payload;
        case FETCH_RELEASE:
            return { ...state, [action.payload.data.id]: action.payload.data };
        default:
            return state;
    }
}
