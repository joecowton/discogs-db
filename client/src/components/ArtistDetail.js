import _ from 'lodash';
import React, { Component } from 'react';
import ReactPlayer from 'react-player';
import { Link, withRouter  } from 'react-router-dom';
import { fetchArtist } from '../actions';
import Payments from './Payments';
import { connect } from 'react-redux';

class ArtistDetail extends Component {
  componentWillMount() {
    const { artist } = this.props.match.params;
    this.props.fetchArtist(artist);
  }

  mapImages(release) {
    return _.map(release.images, image => {
      return <image
        src={image.uri}
        height="200" width="200" padding="10"
        onClick={() => console.log("here")}></image>
    });
  }

  mapArtists(release){return _.map(release.artists, artist => {
      return <div>{artist.name} {artist.join}</div>
    })
  }

  mapVideos(release){return _.map(release.videos, video => {
      return (
        <div className="container" style={{ display: 'inline-grid'}}>
          <div className="row">
            <div className="col-3">
              <ReactPlayer url={video.uri} height="100" width="100%" />
            </div>
          </div>
         </div>
      )
    })
  }

  renderData(data) {

    return (
        <ul key={data.id} className="collection">
          <li className="collection-item avatar">
            <Link to={`/release/${data.id}`}>
              <img src={data.thumb} alt="" className="circle"></img>
            </Link>
            <span className="title">{data.artist}</span>
            <p>{data.title} <br />
               {data.label}
            </p>
          </li>
        </ul>
    )
  }

    render() {
    const { data } = this.props;

    if (!data.length){
      return (
        <div className="progress white">
          <div className="indeterminate green"></div>
        </div>
      )
    }

    return (
      <div>
        <ul className="list-group center">
          {this.props.data.map(this.renderData)}
        </ul>
      </div>
    )
  }
}

function mapStateToProps(state) {
  return { data: state.data }
}

export default connect(mapStateToProps, { fetchArtist })(ArtistDetail);
