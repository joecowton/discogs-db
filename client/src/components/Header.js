import React, { Component } from 'react';
import { BrowserRouter, Router, Link  } from 'react-router-dom';
import { connect } from 'react-redux';
import Payments from './Payments';

class Header extends Component {
  renderContent() {
    switch (this.props.auth) {
      case null:
        return;
      case false:
        return <li><a href="/auth/google">Login</a></li>
      default:
        return [
          <li key="3" style={{ margin: '0 10px' }}>
            Logged in as: {this.props.auth.name}
          </li>,
          <li key="2"><a href="/api/logout">Logout</a></li>
      ]
    }
  }

  render() {
    return (
      <div>
          <nav className="navbar navbar-inverse ">
            <div className="container-fluid" >
              <div className="navbar-logo ">
                <Link to={this.props.auth ? '/releases/all' : '/' }
                  className="left brand-logo" style={{}}>
                  <a>Livity Sound</a>
                </Link>
                <ul className="right">
                  {this.renderContent()}
                </ul>
              </div>
            </div>
          </nav>
      </div>
    )
  }
}

function mapStateToProps({ auth }) {
  return { auth }
}


export default connect(mapStateToProps)(Header);
