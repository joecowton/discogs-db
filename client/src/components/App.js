import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../actions';
import Header from './Header';
import Landing from './Landing';
import ReleaseList from './ReleaseList';
import ReleaseNew from './releaseForm/ReleaseNew';
import ReleaseDetail from './ReleaseDetail';

class App extends Component {
	componentDidMount() {
		this.props.fetchUser();
	}

	render() {
		return (
			<div>
				<BrowserRouter>
					<div className="container">
						<Header location={this.props.location} />
						<Switch>
							<Route exact path="/" component={Landing} />
							<Route path="/releases/all" component={ReleaseList} />
							<Route path="/releases/new" component={ReleaseNew} />
							<Route path="/releases/:id" component={ReleaseDetail} />
						</Switch>
					</div>
				</BrowserRouter>
			</div>
		);
	}
}

export default connect(null, actions)(App);
