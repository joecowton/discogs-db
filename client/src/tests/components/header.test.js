import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Header } from '../../components/Header';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState as featureComponent } from '../../reducers/dataReducer.js';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<Header />', () => {
	it('renders with authenticated props', () => {
		const auth = { name: 'Something Something', GoogleID: 12321312 };
		const wrapper = shallow(<Header {...auth} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('renders without authenticated props', () => {
		const wrapper = shallow(<Header />);

		expect(wrapper).toMatchSnapshot();
	});
});
