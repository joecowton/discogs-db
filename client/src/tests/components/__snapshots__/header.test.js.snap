// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders with authenticated props 1`] = `
<div>
  <nav
    className="navbar navbar-inverse "
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-logo"
      >
        <Link
          replace={false}
          to="/"
        >
          Main
        </Link>
        <ul
          className="right"
        >
          <div
            key="3"
          >
            <li
              key="1"
            >
              <Link
                replace={false}
                to="/"
              >
                Add New
              </Link>
            </li>
            <li
              key="2"
            >
              <a
                href="/api/logout"
              >
                Logout
              </a>
            </li>
          </div>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`<Header /> renders without authenticated props 1`] = `
<div>
  <nav
    className="navbar navbar-inverse "
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-logo"
      >
        <Link
          replace={false}
          to="/"
        >
          Main
        </Link>
        <ul
          className="right"
        >
          <div
            key="3"
          >
            <li
              key="1"
            >
              <Link
                replace={false}
                to="/"
              >
                Add New
              </Link>
            </li>
            <li
              key="2"
            >
              <a
                href="/api/logout"
              >
                Logout
              </a>
            </li>
          </div>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
